### ISSUE TITLE
Error in retraction

### ISSUE THREAD

**Issue by robinloujun** [whuang08/ROPTLIB#13]
Thanks for the great work in Riemannian optimization applications.

I'm new to the library and want to integrate it into my framework. But I'm facing the problem when I compile the library in C++.

- First I tried to work with the released version 0.6. The f2c.h header caused conflict with <complex>, which is already be posted in issue #5 and the "real" and "complex" were modified in commit 240ec9a3ab6c3f8f0d28a905f34a557e1e2d8d1b. But the retraction error arises with the warning `Error in qf retraction` when I run the examples TestSimpleExample and TestProductExample. I didn't make any change in the test source files.

- Then I turned to the later version (version 0.7 and the master branch) but the seperate checking system seems to be disabled and all the checks are integrated into DriverCpp (it leads to multiple definition of main() when try to compile TestSimpleExample, TestProductExample etc. ). But there is still such errors in almost all the tested methods. So I'm a bit confused.

- Besides, in the updated user manual provided in issue #6 , the installation instruction for FFTW3 should be (with an additional dash)
```shell
sudo apt-get install libfftw3-*
```

**Comment by whuang08** [comment]
See my responses below:
*) Modifying Line 10 in f2c.h from "typedef long int integer;" to "typedef int integer;" should fix the problem. The "long int" works in Windows and I usually work on Windows. Therefore, I did not notice this issue. I suppose you are working on Ubuntu.

*) Sorry for the confusion. The one on the master branch is still under development. Since I'm working on Windows system with Matlab and C++ environments usually, I did not get chance to test on Ubuntu and MAC systems yet. After making a fully test, I will release another version. I think with the updates above, version 0.6 should work for you. But if you want to use the version on the master branch, I suggest you compile TestStieBrocket or any other test files that do not include main() functions. I think TestSimpleExample and TestProductExample are the only two tests that still have main() function.

*) Thanks. I will keep this in mind, use it in the next fully test, and update user manual if it works.
